packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+    const res = arePeriodsWithinRange([resource.onsetPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetRange) {
+    const res = areRangesWithinRange([resource.onsetRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetString) {
+    const res = isDateWithinDateRange(resource.onsetString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.onsetDateTime) {
+    const res = isDateWithinDateRange(resource.onsetDateTime, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementPeriod) {
+    const res = arePeriodsWithinRange([resource.abatementPeriod], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementRange) {
+    const res = areRangesWithinRange([resource.abatementRange], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementString) {
+    const res = isDateWithinDateRange(resource.abatementString, range);
+    if (res !== undefined) return res;
+  }
+  if (resource.abatementDateTime) {
+    const res = isDateWithinDateRange(resource.abatementDateTime, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isEncounterWithinDateRange(resource: Encounter, range: DateRange): boolean {
+  if (resource.period) {
+    const res = arePeriodsWithinRange([resource.period], range);
+    if (res !== undefined) return res;
+  }
+  if (resource.location) {
+    const periods = resource.location.map(l => l.period);
+    const res = arePeriodsWithinRange(periods, range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isFamilyMemberHistoryWithinDateRange(
+  resource: FamilyMemberHistory,
+  range: DateRange
+): boolean {
+  if (resource.condition) {
+    const periods = resource.condition.map(c => c.onsetPeriod);
+    const arePeriods = arePeriodsWithinRange(periods, range);
+    if (arePeriods !== undefined) return arePeriods;
+
+    const ranges = resource.condition.map(c => c.onsetRange);
+    const areRanges = areRangesWithinRange(ranges, range);
+    if (areRanges !== undefined) return areRanges;
+
+    const dates = resource.condition.flatMap(c => safeDate(c.onsetString) ?? []);
+    const areDates = areDatesWithinRange(dates, range);
+    if (areDates !== undefined) return areDates;
+  }
+  if (resource.date) {
+    const res = areDatesWithinRange([resource.date], range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isDiagnosticReportWithinDateRange(resource: DiagnosticReport, range: DateRange): boolean {
+  const dates = getDatesFromEffectiveDateTimeOrPeriod(resource);
+  const res = areDatesWithinRange(dates, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isDocumentReferenceWithinDateRange(
+  resource: DocumentReference,
+  range: DateRange
+): boolean {
+  const res = isPeriodWithinRange(resource.context?.period, range);
```

Comment:
missing resource.date which is common place we have the date

Reply:
I intentionally chose not to add resource.date, since that is just the date when the resource was created, it doesn't seem related to the clinical information the customers are trying to filter for.

Thoughts?

Reply:
I checked a bunch of DocumentReferences and saw empirically that that is where there is a often the date. Maybe worth investigating quickly where we put dates when we construct the document References

Reply:
Adding this to a ticket as it seems out of scope of this PR: https://github.com/metriport/metriport-internal/issues/2241

Cool?